src/LanguageServer/Protocol/Handler/Extensions/ExtensionRegisterResponse.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Text.Json.Serialization;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.Extensions;
+
+/// <summary>
+/// Response for the roslyn/extensionRegister request.
+/// </summary>
+/// <param name="WorkspaceMessageHandlers">Names of the registered non-document-specific extension message handlers.</param>
+/// <param name="DocumentMessageHandlers">Names of the registered document-specific extension message handlers.</param>
+internal readonly record struct ExtensionRegisterResponse(
+    [property: JsonPropertyName("workspaceMessageHandlers")] string[] WorkspaceMessageHandlers,
+    [property: JsonPropertyName("documentMessageHandlers")] string[] DocumentMessageHandlers);
```

Comment:
ImmutalbeArrays.

Reply:
Does serialization work w/ ImmutableArrays?  Typically we've been using arrays for messages

Reply:
yes.  serialization works with IAs.  we use it all over the place for OOP.

Reply:
I'll do ImmutableArrays and go back to arrays in case it fails when I test it